driver/gl/raster_cache.go

```diff
@@ -0,0 +1,66 @@
+package gl
+
+import (
+	"image"
+	"os"
+	"sync"
+	"time"
+
+	"fyne.io/fyne"
+)
+
+type rasterInfo struct {
+	pix     *image.RGBA
+	w, h    int
+	expires time.Time
+}
+
+var cacheDuration = (time.Minute * 5)
+var rasters = make(map[fyne.Resource]*rasterInfo)
+var rasterMutex sync.RWMutex
+
+func init() {
+	if t, err := time.ParseDuration(os.Getenv("FYNE_CACHE")); err == nil {
+		cacheDuration = t
+	}
+
+	go func() {
+		delay := cacheDuration / 2
+		if delay < time.Second {
+			delay = time.Second
+		}
+		for {
+			time.Sleep(delay)
+			now := time.Now()
+			rasterMutex.Lock()
+			for k, v := range rasters {
+				if v.expires.Before(now) {
+					delete(rasters, k)
+				}
+			}
+			rasterMutex.Unlock()
+		}
+	}()
+}
+
+func svgCacheGet(res fyne.Resource, w int, h int) *image.RGBA {
+	rasterMutex.RLock()
+	defer rasterMutex.RUnlock()
+	v, ok := rasters[res]
+	if !ok || v == nil || v.w != w || v.h != h {
+		return nil
+	}
+	v.expires = time.Now().Add(cacheDuration)
+	return v.pix
+}
+
+func svgCachePut(res fyne.Resource, pix *image.RGBA, w int, h int) {
+	rasterMutex.Lock()
+	defer rasterMutex.Unlock()
+	rasters[res] = &rasterInfo{
```

Comment:
The `fyne.Resource` type is an interface, and is not guaranteed to be comparable. If an implementation directly contains a slice (not a pointer to something containing a slice), this will panic. I would therefore add a defer containing a recover before this.

Reply:
maybe we can use res.Name() instead?

Reply:
didnt see that.  If its an interface, then its not a good thing to be using as a map key, totally agree.

Ive added the recover() for now, but a more robust solution might be to extend the resource interface to include a uniquely identifier, the value of which would make a better map key.

Thats a larger change.